### DISCUSSION TITLE
issue: Stale form error when using resolver.

### DISCUSSION THREAD

**Discussion by deskoh** [react-hook-form/react-hook-form#10920] [Q&A]
### Version Number

7.45.3

### Codesandbox/Expo snack

https://codesandbox.io/s/stale-formstate-error-v2qjqd

### Steps to reproduce

1. Click first text field and change value to 50.
2. Resolver runs. See the error message `Sum with B must be 10 (1)`
3. Click second text field and change value to 50.
4. Click anywhere outside the text field to trigger `onBlur`
5. Resolver runs. See error message `Sum with A must be 10 (2)`
6. Click 'Trigger All' button to trigger resolver again.
7. See both error message `Sum with A must be 10 (3)` / `Sum with B must be 10 (3)`. [Expected]
8. Click 'Set Error' call `setError` on both text fields.
9. See both error message `Error for A` / `Error for B`. [Expected]

### Expected behaviour

The form validation `mode` is set to `onBlur`.

As the error state for both inputs are subscribed, it seemed that errors set using the resolver doesn't trigger an update as seen in step 2 and step 5.

I would expect Step 2 to see error message (`Sum with A must be 10 (1)`) for second text field and Step 5 to see updated error message (`Sum with B must be 10 (2)`) for first text field 

Both errors are updated only when `methods.trigger` or `setError`s are called in step 6 and 8 respectively.

P.S. Behavior is similar if mode is set to `onChange` or `all`.

My actual use case requires complex validation dependencies using controlled input implemented using `useController`, so I cannot use `deps` option in `register`. One solution is to watch all values and call `methods.trigger` when the values are change. Is that the most optimal way?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
Please, refer to #8170 for the API design reasoning.

> One solution is to watch all values and call methods.trigger when the values are change. Is that the most optimal way?

The React way would be to call the `trigger` method as an event callback instead of in a `useEffect`. Check out [this React docs page](https://react.dev/learn/you-might-not-need-an-effect#how-to-remove-unnecessary-effects) on the topic.

**Comment by deskoh** [comment]
@Moshyfawn thanks for the clarifications.
Are you able to provide some guidance what event callback can be hooked into to call the `trigger` method?

Are you referring to the individual fields `onBlur` or `onChange` events?

**Reply by trile-coalition** [reply]
@deskoh Were you able to figure out this issue?

**Reply by deskoh** [reply]
Yeah... Calling `trigger` in `onBlur` seemed to work.
